scipy/stats/_distn_infrastructure.py

```diff
@@ -3286,6 +3267,128 @@ def _iter_chunked(x0, x1, chunksize=4, inc=1):
         yield supp
 
 
+class rv_sample(rv_discrete):
+    """A 'sample' dicrete distribution defined by the support and values.
+
+       The ctor ignores most of the values, only needs the `values` argument.
+    """
+    def __init__(self, a=0, b=inf, name=None, badvalue=None,
+                 moment_tol=1e-8, values=None, inc=1, longname=None,
+                 shapes=None, extradoc=None, seed=None):
+
+        super(rv_discrete, self).__init__(seed)
+
+        if values is None:
+            raise ValueError("rv_sample.__init__(..., values=None,...)")
+
+        # cf generic freeze
+        self._ctor_param = dict(
+            a=a, b=b, name=name, badvalue=badvalue,
+            moment_tol=moment_tol, values=values, inc=inc,
+            longname=longname, shapes=shapes, extradoc=extradoc, seed=seed)
+
+        if badvalue is None:
+            badvalue = nan
+        if name is None:
+            name = 'Distribution'
+        self.badvalue = badvalue
+        self.name = name
+        self.moment_tol = moment_tol
+        self.inc = inc
+        self._cdfvec = vectorize(self._cdf_single, otypes='d')
+        self.vecentropy = vectorize(self._entropy)
+        self.shapes = shapes
+        self.extradoc = extradoc
+
+        self.xk, self.pk = values
+        self.return_integers = 0
+        indx = argsort(ravel(self.xk))
+        self.xk = take(ravel(self.xk), indx, 0)
+        self.pk = take(ravel(self.pk), indx, 0)
+        self.a = self.xk[0]
+        self.b = self.xk[-1]
+        self.P = dict(zip(self.xk, self.pk))
+        self.qvals = np.cumsum(self.pk, axis=0)
+        self.F = dict(zip(self.xk, self.qvals))
+        decreasing_keys = sorted(self.F.keys(), reverse=True)
+        self.Finv = dict((self.F[k], k) for k in decreasing_keys)
+        self._ppf = instancemethod(vectorize(_drv_ppf, otypes='d'),
+                                   self, rv_discrete)
+        self._pmf = instancemethod(vectorize(_drv_pmf, otypes='d'),
+                                   self, rv_discrete)
+        self._cdf = instancemethod(vectorize(_drv_cdf, otypes='d'),
+                                   self, rv_discrete)
```

Comment:
those can be added as methods directly
as instancemethods they should use rv_sample instead of rv_discrete, I think